### ISSUE TITLE
Warn if `?.` operator is being used on non-nullable properties

### ISSUE THREAD

**Issue by jmarolf** [dotnet/roslyn#53703]
Today we do not warn about unnecessary `?.` operators

**Comment by RikkiGibson** [comment]
It feels like the diagnostic given for `a?.b` should be similar to the diagnostic given in the same context for `a != null ? a.b : null`. I don't expect the compiler to give this diagnostic, but the IDE may decide to do so to gray out what it thinks are unnecessary null tests. It would probably need to be paired with some heuristics like "don't gray out null tests on parameters/properties/etc in public API."

**Comment by jmarolf** [comment]
## Design Review

**Discussion:**
 - Nullable brought up a lot of small detail-oriented pieces to discuss. In general there is a concern about nullable obliviousness that may be necessary for this sort of diagnostic to be do-able at all. Joey brought up that we would likely want to color these null checks as faded-out or “unnecessary”. DavidW pointed out that this sort of analysis should be done for all unnecessary null checks not just the Elvis Operator.

**Conclusion:**
 - implement prototype in roslyn-analyzers and run against dotnet/runtime, roslyn etc

**Comment by RikkiGibson** [comment]
It feels like the design needs to be aware that it will miss some cases. e.g.

```cs
var x = ObliviousProp;
x?.M();
```

**Comment by CyrusNajmabadi** [comment]
I'm very wary about a warning here. It seems to go against our philosophy around nrt. Specifically, we recognize that annotations may be incorrect and that user code may be written to still be resilient to unintended nulls. 

This feels like this collides with that, despite the null check being the true safeguard and the annotation a weak second.
